downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -318,7 +318,7 @@ enum class VcsHost(
         private val SVN_BRANCH_OR_TAG_PATTERN = Regex("(.*svn.*)/(branches|tags)/([^/]+)/?(.*)")
         private val SVN_TRUNK_PATTERN = Regex("(.*svn.*)/(trunk)/?(.*)")
         private val GIT_REVISION_FRAGMENT = Regex("git.+#[a-fA-F0-9]{7,}")
-        private val GIT_PROJECT_NAME = Regex("/([^/]+)\\.git")
+        private val GIT_PROJECT_NAME = Regex("/([^/]+?)(\\.git)?$")
```

Comment:
How an we be sure it's a *Git* project name now if the ".git" suffix is optional, and also no other sign of Git is being checked for?

Reply:
Excellent question. There is no way.

I found this with `pubspec.yml`, where Git is the only supported VCS at the moment: https://dart.dev/tools/pub/dependencies#dependency-sources

Should I create `VcsHost.getGitProject()` function which can make such bold assumption?

Reply:
> I found this with `pubspec.yml`, where Git is the only supported VCS at the moment:

Wouldn't the better solution then be to hard-code `VcsType.GIT` for every `VcsInfo` that the Pub analyzer creates?

Reply:
That is already done: https://github.com/oss-review-toolkit/ort/blob/main/plugins/package-managers/pub/src/main/kotlin/Pub.kt

The problem is that https://github.com/oss-review-toolkit/ort/blob/main/plugins/package-managers/pub/src/main/kotlin/utils/PubCacheReader.kt#L93 uses `VcsHost.getProject()` helper function in order to extract the _repoName.

Since Pub does things differently, maybe I should remove the whole `VcsHost.getProject()` and replicate what Pub does in https://github.com/dart-lang/pub/blob/ea4a1c854690d3abceb92c8cc2c6454470f9d5a7/lib/src/source/git.dart#L735
```
  String _repoName(GitDescription description) {
    var name = p.url.basename(description.url);
    if (name.endsWith('.git')) {
      name = name.substring(0, name.length - '.git'.length);
    }
    name = name.replaceAll(RegExp('[^a-zA-Z0-9._-]'), '_');
    // Shorten name to 50 chars for sanity.
    if (name.length > 50) {
      name = name.substring(0, 50);
    }
    return name;
  }
```

Reply:
I will do another PR for `PubCacheReader.kt`.